web/src/components/PoisMobile.tsx

```diff
@@ -160,11 +172,16 @@ const PoisMobile = ({
         setBottomActionSheetHeight={setBottomActionSheetHeight}
         direction={direction}>
         {currentFeatureOnMap && isBottomActionSheetFullScreen && (
-          <PoiGoBack goBack={() => selectPoiFeatureInList(null)} direction={direction} viewportSmall t={t} />
+          <PoiGoBack
+            goBack={() => selectPoiFeatureInList(null)}
+            direction={direction}
+            viewportSmall
+            text={t('detailsHeader')}
+          />
         )}
         <ListContainer>
-          {currentPoi && currentPoiFeature ? (
-            <PoiDetails poi={currentPoi} feature={currentPoiFeature} direction={direction} t={t} />
+          {currentPoi ? (
+            <PoiDetails poi={currentPoi} feature={currentPoi.getFeature(userLocation)} direction={direction} t={t} />
```

Comment:
If this is the reason why you did all the changes, the following should still work: 
```
const currentPoiFeature = currentFeatureOnMap?.properties.pois.find(poi => poi.slug === currentPoi?.slug)
```

Not sure if that includes the user location though.